plugins/permission-react/src/components/WithPermission.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Permission } from '@backstage/plugin-permission-common';
+import React, { PropsWithChildren } from 'react';
+import { usePermission } from '../hooks';
+
+/**
+ * React Higher-Order Component that provides conditional rendering after an authorization check.
+ *
+ * Usage:
+ * ```
+ * <WithPermission permission={somePermission}>
+ *   <MyComponent />
+ * </WithPermission>
+ * ```
+ * MyComponent will only render if the user is authorized to perform somePermission.
+ * @public
+ */
+export const WithPermission = ({
```

Comment:
Would be `withPermission` right? Tbh I'm not sure we need to export this though as we generally don't supply HOCs that are trivial to implement in terms of hooks. It does add to the API surface and I'm not sure it's that useful tbh.

Reply:
I write ^ but then I look at the docs and realize that this isn't a HOC at all. We don't have a lot of usage of these kind of child render blocking components, but there are a couple in the internal project and we call them barriers there. What do you think about `PermissionBarrier`?

Reply:
Good catch on it not being an HOC - that's my mistake. How about `PermissionBoundary`? It's basically like an Error Boundary (or a Suspense, for that matter) after all.

Reply:
Hmm, so the reasoning for not using "boundary" is that it seems like that's a slightly different thing in React. Boundaries, that is `ErrorBoundary` and `Suspense`, are both catching things that are bubbling up from children, while what we're doing in this case is to not render children unless certain criteria are met.

So thinking here is that "barriers" interact with information as it's coming down the element tree, while "boundaries" interact with information coming back up the tree.

Barrier is still something we made up though, and would be happy to find some established naming around these things x)
Totally up for calling it `PermissionBoundary` too but ^ is the current reasoning anyway.

Reply:
[Here](https://github.com/backstage/backstage/blob/mob/authorization-framework/packages/app/src/components/Root/Root.tsx#L92)'s the usage of this component in the PRFC branch, in case it's useful when considering these names. I agree that `Boundary` could perhaps lead folks to imagine there's something bubbling up. 

More broadly: if this isn't a pattern we've used much for conditional rendering in Backstage open source to date, then perhaps we should skip this component for now and stick with using the `usePermission` hook directly?

Reply:
The discussion is pushing me towards leaving it out for now. If we end up seeing more and more cases where we pull in the hook just to conditionally render, we can circle back to this.